### DISCUSSION TITLE
How to Autostart files,programs in LXQt 1.2.0,  Qt 5.15.8, debian-live-12.6.0-amd64-lxqt.iso?

### DISCUSSION THREAD

**Discussion by rajibando** [lxqt/lxqt#2606] [Q&A] [answered]
I tried by creating an executable file in the directory, `/home/user/.config/autostart`.
File name: `auto.desktop`, with the following codelines:

______________________[Desktop Entry]
Type=Application
Name=Startup Commands
Exec=sh -c '
  featherpad &
  pcmanfm-qt &
  '
Hidden=false
X-LXQt-Autostart-enabled=true
Name=auto.desktop______________________

But doesn't work.

There is no reference to the use of \S for multi-line Exec commands. The specification expects Exec to be a single-line command. For multiple commands, it recommends wrapping them in a shell script and then calling that script from the Exec key.

Where am I going wrong?

**Comment by isf63** [comment]
Just a wild guess: do you need `\`s for multiline in Exec?

Here's the desktop entry spec: https://specifications.freedesktop.org/desktop-entry-spec/latest/

**Comment by stefonarch** [comment]
You don't need to fiddle, use Session Settings > Autostart > Add > enter the Name and the executable (fpad, pcmanfm-qt, firefox, whatever)

Those files don't need to be executable, btw.

**Reply by rajibando** [reply]
> ...use Session Settings > Autostart > Add > enter the Name and the executable (fpad, pcmanfm-qt, firefox, whatever) 
> Those files don't need to be executable, btw.

😄
Is a **Maintainer** strictly restricted by word-limits, Mr. Stefonarch, of the Team Maintainers?!

Hope that the above line is treated in its true spirit of harmless humour, not satire.

Should the line:
 "enter the Name and the executable"
  instead be:
  "enter the Name of the executable"?
  
 Otherwise, the second line, "Those files don't need to be executable, btw." doesn't logically support the first line.

Next, what about commands, such as:
`qterminal -e xset dpms force off &`,
messages such as:
`xmessage -center "Please check if the charger is connected" &
  beep -f 1000 -l 500 -r 3 &`
and bash script files.

**Reply by stefonarch** [reply, marked as answer]
>Otherwise, the second line, "Those files don't need to be executable, btw." doesn't logically support the first line.

I referred to the .desktop files, not to the executable in them with that.

For entering complex commands you need to include them between quotes well. Otherwise you can enter the path to the script containing them, using the "Search" button in the "Add dialog".

**Reply by rajibando** [reply]
> I referred to the .desktop files, not to the executable in them with that.

Yes, when I tried your earlier suggestion:

> You don't need to fiddle, use Session Settings > Autostart > Add > enter the Name and the executable (fpad, pcmanfm-qt, firefox, whatever)

I observed that respective filenames with .desktop file extension are individually created within the autostart directory when I followed your above-quoted advice. These created files don't need to be executable.
Nice!

> For entering complex commands you need to include them between quotes well. Otherwise you can enter the path to the script containing them, using the "Search" button in the "Add dialog".

Yes, truly. within the .desktop files. complex commands need effort to be constructed. Better is to call scripts from within the individual files with .desktop file extensions.

So your method is neat. My project is accomplished.

Thanks. Your two answers need to be combined to form the complete answer. How to mark your two answers as one complete answer? Because I can't.

This aspect could be looked into.
